v3-certificate/generate-certificate

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+#set -e
+
+# Checking openssl
+if [[ -z $(which openssl) ]]; then
+  echo "Please download openssl and make sure it is available in your PATH variable"
+  echo "Homepage: https://www.openssl.org/"
+  exit 1
+fi
+
+# Checking jq
+if [[ -z $(which jq) ]]; then
+  echo "Please download jq and make sure it is available in your PATH variable"
+  echo "jq is used to parse JSON elements"
+  echo "Homepage: https://stedolan.github.io/jq/"
+  exit 1
+fi
+
+# FusionAuth URL
+read -r -p "Your FusionAuth instance URL (with scheme): " FA_URL
+regex='^https?:\/\/[^\/]+(:[0-9]+\/)?$'
+if ! [[ $FA_URL =~ $regex ]]; then
+  echo "FusionAuth URL is not valid. Please make sure you have included the scheme (http:// or https://)."
+  exit 2
+fi
+
+# Extracting domain from URL
+FA_DOMAIN=$(echo "$FA_URL" | awk -F/ '{print $3}')
+if [[ -z "$FA_DOMAIN" ]]; then
+  echo "FusionAuth URL is not valid. Please make sure you have included the scheme (http:// or https://)."
+  exit 2
+fi
+
+# FusionAuth API Key
+read -r -p "API Key with /api/key/import endpoint:      " API_KEY
+API_KEY_LENGTH=${#API_KEY}
+if [ "$API_KEY_LENGTH" -ne 56 ]; then
+  echo "Your API Key seems invalid. Please make sure you copied it correctly."
+  exit 2
+fi
+
+# Key name
+read -r -p "Name for the generated key:                 " NAME
+if [[ -z "$NAME" ]]; then
+  echo "Please provide a valid name for the key."
+  exit 2
+fi
+
+# Creating temporary files for public and private keys
+PRIVATE_KEY_FILE=$(mktemp)
+if [[ -z "$PRIVATE_KEY_FILE" ]]; then
+  echo "Cannot create private key. Please make sure your temporary directory is writable."
+  exit 3
+fi
+PUBLIC_KEY_FILE=$(mktemp)
+if [[ -z "$PUBLIC_KEY_FILE" ]]; then
+  echo "Cannot create public key. Please make sure your temporary directory is writable."
+  exit 3
+fi
+
+# Creating certificate
+RSA_LENGTH=2048
+echo "Generating certificate..."
+if ! openssl req -x509 -newkey "rsa:$RSA_LENGTH" -nodes -keyout "$PRIVATE_KEY_FILE" -out "$PUBLIC_KEY_FILE" -sha256 -days 365 -subj "/CN=$FA_DOMAIN/"; then
```

Comment:
I would make the lifetime 3650 if possible.

10 years is what we do for other created keys.

Reply:
Do we need to explicitly set this to version 3 or is that default for openssl?

Reply:
> I would make the lifetime 3650 if possible.
Done

> Do we need to explicitly set this to version 3 or is that default for openssl?
It is the default and available since 2008, so I think we are good. There is no flag to force the version besides adding specific x509v3 extensions to the command, but I think it could cause more harm as it would expect some things from the `openssl.cnf` file...